codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -753,9 +753,23 @@ Finds the earliest queue item across all registers and then returns that
 queue item with an update fetch state.
 */
 let getEarliestEvent = (self: t) => {
-  let registerWithEarliestQItem = self.baseRegister->findRegisterIdWithEarliestQueueItem
-  //Can safely unwrap here since the id is returned from self and so is guarenteed to exist
-  self.baseRegister->popQItemAtRegisterId(~id=registerWithEarliestQItem)->Utils.unwrapResultExn
+  if self.pendingDynamicContracts->Utils.Array.isEmpty {
+    let registerWithEarliestQItem = self.baseRegister->findRegisterIdWithEarliestQueueItem
+    //Can safely unwrap here since the id is returned from self and so is guarenteed to exist
+    self.baseRegister->popQItemAtRegisterId(~id=registerWithEarliestQItem)->Utils.unwrapResultExn
+  } else {
+    let earliestPendingDynamicContractBlockNumber =
+      self.pendingDynamicContracts
+      ->Array.reduce(None, (accum, dynamicContractRegistration) => {
+        switch accum {
+        | None => Some(dynamicContractRegistration.registeringEventBlockNumber)
+        | Some(accumBlockNumber) =>
+          min(accumBlockNumber, dynamicContractRegistration.registeringEventBlockNumber)->Some
+        }
+      })
+      ->Option.getExn
+    NoItem({blockTimestamp: 0, blockNumber: earliestPendingDynamicContractBlockNumber - 1})
```

Comment:
In this case, the pending dynamic contracts do not yet have a register with a "latestFetchedBlock", so it was not being compared with the other registers. 

Perhaps an improvement to this would be to compare the final "NoItem" returned with whatever was the earliest item in the registers.